woocommerce/woocommerce
# Add menu item to publish button to schedule product updates

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40376



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the `New product editor` is enabled and the Prepublish sidebar (`product-pre-publish-modal`) is on under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Go to `Products` > `Add New` and add a `Name` to the product.
3. At the top right corner of the screen, the `Publish` button should have a dropdown menu to the right like. 
<img width="719" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/4da84d12-526f-4010-b946-77b64e2e457a">

4. When clicking it a new item `Schedule publish` should be shown in the list. 
<img width="708" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/636abcf1-0914-45b6-8760-69f202c060aa">

5. After clicking the `Schedule publish` item a new `Schedule product` modal should be shown. 
<img width="559" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/24213db1-557e-4480-ba23-eba37d65de0a">

6. Selecting a date after your current date will schedule a product to be published when clicking the `Schedule` button from the modal.
7. After that a notice should be shown showing the date the product is scheduled for and the `Publish` button should new reads `Schedule` like 
<img width="529" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/30f8a4fc-1138-4520-9d90-a286399be4f4">
<img width="457" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/6d4e3062-b605-45d5-9d1f-e312a4b3995d">

8. Clicking again the dropdown button next to the `Schedule` button should list new two items `Publish now` and `Edit schedule` 
<img width="375" alt="image" src="https://github.com/woocommerce/woocommerce/assets/13334210/35e8f329-5bc8-44e3-8e7d-0173e8ba1254">

9. `Publish now` should publish the product right away whereas `Edit schedule` should open the `Schedule product` modal again to modify the product date.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
